### PR TITLE
Add comment field to feedback panel

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add comment field to feedback panel

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: assets/js/listener.js [hunk 1]
```diff
@@ -17,17 +21,48 @@ document.addEventListener('click', function(event) {
...
+    const text = this.value.trim();
+    const sendButton = document.getElementById('send');
+    
+    if (!document.getElementById('yes').checked && !document.getElementById('no').checked) {
+        text.length > 0 ? sendButton.disabled = false : sendButton.disabled = true;
+    }
+
+    // calculate the number of characters remaining
+    counter = 350 - this.value.length;
+    document.getElementById('num-chars').innerText = counter + " characters left";
```
Comment (AMoo-Miki) [inline assets/js/listener.js:34]: One problem we would have is if a user enters a text and does a refresh, the page would reload with the text in the textfield and a disabled button and the message saying 350 chars to go.

everything being done in the listener should be moved in to a function and that function should be used for the listener and also just called when the page loads (right where the listeners are being added is fine).

File: assets/js/listener.js [hunk 2]
```diff
@@ -17,17 +21,48 @@ document.addEventListener('click', function(event) {
     }
 });
 
+document.getElementById("comment").addEventListener("input", function() {
+    const text = this.value.trim();
+    const sendButton = document.getElementById('send');
+    
+    if (!document.getElementById('yes').checked && !document.getElementById('no').checked) {
```
Comment (AMoo-Miki) [inline assets/js/listener.js:28]: It is wasteful to ask the document to go find these elements with each keystroke. We should store these references to elements in variables outside the function and use those variables instead.

We should do the same to all other functions too.

File: assets/js/listener.js [hunk 3]
```diff
@@ -17,17 +21,48 @@ document.addEventListener('click', function(event) {
...
+        helpful = 'yes';
+    }
+    else if (document.getElementById('no').checked) {
+        helpful = 'no';
+    }
+    
+    let comment = document.getElementById('comment').value.trim();
+    if(comment.length == 0) {
+        comment = 'none';
+    }
```
Comment (AMoo-Miki) [inline assets/js/listener.js:49]: We should validate that we have enough data to proceed.
```
if (helpful === 'none' && comment === 'none') return;

File: assets/js/listener.js (hunk 1 above)
Comment (kolchfa-aws) [reply assets/js/listener.js:34]: That was one of the requirements from UX: the users should be able to reload the page and get back to the initial state, where they would be able to submit feedback again.

File: assets/js/listener.js (hunk 2 above)
Comment (kolchfa-aws) [reply assets/js/listener.js]: Thank you! Refactored.

**Review by carolxob — APPROVED** [review]
LGTM!

File: _includes/feedback.html [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
+        <label class="feedback-radio" for="yes">&#10004; Yes</label> 
+        <input id="no" type="radio" name="radio-button" value="no" class="feedback-button">
+        <label class="feedback-radio" for="no">&#10006; No</label>
     </div>
+    <div class="feedback-text-header">Tell us why</div>
+    <textarea id="comment" class="feedback-text" placeholder="Enter comment" maxlength="350"></textarea>
+    <div class="text-small num-chars" id="num-chars">350 characters left</div>
+    <button id="send" class="send-button" disabled>Send</button>
+    <p class="text-small text-grey-dk-100 hidden" id="thank-you">Thank you for your feedback!</p>
     <p class="text-small text-grey-dk-100">Want to contribute? <a class="feedback-edit" target="_blank" href="https://github.com/opensearch-project/documentation-website/edit/main/{{ page.path }}">Edit this page</a> or <a class="feedback-issue" target="_blank" href="https://github.com/opensearch-project/documentation-website/issues/new?assignees=&labels=untriaged&template=issue_template.md&title=%5BDOC%5D">create an issue</a>.</p>
```
Comment (hdhalter) [inline _includes/feedback.html:14]: 'Edit this page' brings me to a 404 page. Should this link work on staging? 'Create an issue' works fine.

**Review by hdhalter — APPROVED** [review]
Looks great!! Just one question about the Edit this page link.

File: _includes/feedback.html (hunk 4 above)
Comment (kolchfa-aws) [reply _includes/feedback.html:14]: @hdhalter Are you doing this on your local system? And what page are you doing it from?

File: assets/js/listener.js (hunk 1 above)
Comment (kolchfa-aws) [reply assets/js/listener.js]: Implemented as suggested in the most recent commit.

File: _includes/banner.html [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <section class="os-banner">
-    <header class="banner-head" style="background-image: url('{{site.url}}{{site.baseurl}}/images/graphic-bg.jpg')">
+    <header class="banner-head" style="background-image: url('{{site.url}}{{site.baseurl}}/images/graphic-bg.png'); background-size: cover;">
```
Comment (AMoo-Miki) [inline _includes/banner.html:2]: any reason why we should not add the background-size to CSS?

**Review by AMoo-Miki — APPROVED** [review]

File: _includes/banner.html (hunk 5 above)
Comment (kolchfa-aws) [reply _includes/banner.html:2]: No reason. Good catch, thank you. I'll update in another PR.

Comment (kolchfa-aws) [reply _includes/banner.html:2]: Implemented in #2962

**Pull request merged**